src/Functions/formatDateTime.cpp

```diff
@@ -385,32 +381,31 @@ class FunctionFormatDateTimeImpl : public IFunction
                     {
                         using FromDataType = std::decay_t<decltype(type)>;
                         if (!(res = executeType<FromDataType>(arguments, result_type)))
-                            throw Exception(
-                                "Illegal column " + arguments[0].column->getName() + " of function " + getName()
-                                    + ", must be Integer or DateTime.",
-                                ErrorCodes::ILLEGAL_COLUMN);
+                            throw Exception(ErrorCodes::ILLEGAL_COLUMN,
+                                "Illegal column {} of function {}, must be Integer or DateTime.",
```

Comment:
"[...] must be be Integer, Date, Date32, DateTime or DateTime64"

Reply:
Done